packages/engine/Source/Core/OrientedBoundingBox.js

```diff
@@ -714,6 +710,168 @@ OrientedBoundingBox.intersectPlane = function (box, plane) {
   return Intersect.INTERSECTING;
 };
 
+const scratchInverse = new Matrix4();
+const scratchNormal = new Cartesian3();
+const scratchRay = new Ray();
+
+/**
+ * Computes the nearest intersection along the ray with the oriented bounding box.
+ *
+ * @param {OrientedBoundingBox} box The oriented bounding box to test.
+ * @param {Ray} ray The ray to test against.
+ * @param {Cartesian3} [result] The nearest intersection along the provided ray with the oriented bounding box.
+ * @returns {Cartesian3|undefined} The nearest intersection along the provided ray with the oriented bounding box, or <code>undefined</code> if there is no intersection.
+ */
+OrientedBoundingBox.intersectRay = function (box, ray, result) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.defined("box", box);
+  Check.defined("ray", ray);
+  //>>includeEnd('debug');
+
+  const transformedRay = scratchRay;
+  transformedRay.origin = Cartesian3.subtract(
+    ray.origin,
+    box.center,
+    transformedRay.origin
+  );
+  transformedRay.direction = Cartesian3.clone(ray.direction);
```

Comment:
transformedRay.direction = Cartesian3.clone(ray.direction, transformedRay.direction);
to avoid allocation?

BTW: I think the X/Y/Z cases below could go into some

  // Implementation based on https://education.siggraph.org/static/HyperGraph/raytrace/rtinter3.htm
  if (!updateRange(Cartesian3.UNIT_X, origin.x)) {
    return;
  }
  if (!updateRange(Cartesian3.UNIT_Y, origin.y)) {
    return;
  }
  if (!updateRange(Cartesian3.UNIT_Z, origin.z)) {
    return;
  }

but I probably overlooked something here.

Reply:
Since it's a fairly short repetition, I left it unfolded for better readability.